elastic/shared/runners/remote_cluster.py

```diff
@@ -162,9 +173,9 @@ async def _follow_indices(self, params, source_cluster_name, source_cluster_clie
                 await following_cluster_client.ccr.follow(
                     index=index, wait_for_active_shards="1", body=follow_body, request_timeout=request_timeout
                 )
-            except ElasticsearchException as e:
+            except (ApiError, TransportError, ElasticsearchException) as e:
                 msg = f"Failed to follow index [{index}] from [{source_cluster_name}] on [{following_cluster_name}]; [{e}]"
-                raise BaseException(msg)
+                raise e(msg)
```

Comment:
I realize that this was working this way before this pull request, but by raising a new exception we're losing the traceback from the previous exception here. The best solution would be to use [add_note](https://docs.python.org/3/tutorial/errors.html#enriching-exceptions-with-notes) but it's only available starting with Python 3.11. The next best thing is to remove `; [{e}]` from `msg` and use [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining):

```suggestion
                raise Exception(msg) from e
```

That way we keep the context from the previous exception while still raising our own exception. Now we have to think about the exception we want to raise. `BaseException` was a bad choice in my opinion as it prevents Ctrl-C from having any effect. `Exception` works but is very broad. Then there's  `RallyError` or one of its subclasses.

Reply:
Thanks for the suggestions! I think I've covered all bases in https://github.com/elastic/rally-tracks/pull/380/commits/e1ba2306babf0ca7cc5091433b8252fafceeeb74.